File: php/src/Utils/Compression.php
<?php

declare(strict_types=1);

namespace Royaltics\ErrorTracker\Utils;

final class Compression
{
    public static function compressAndEncode(string $data): string
    {
        $compressed = gzencode($data, 9);
        
        if ($compressed === false) {
            throw new \RuntimeException('Failed to compress data');
        }
        
        return base64_encode($compressed);
    }
}

File: csharp/Utils/Compression.cs
using System.IO.Compression;
using System.Text;

namespace Royaltics.ErrorTracker.Utils;

public static class Compression
{
    public static string CompressAndEncode(string data)
    {
        var bytes = Encoding.UTF8.GetBytes(data);
        
        using var outputStream = new MemoryStream();
        using (var gzipStream = new GZipStream(outputStream, CompressionMode.Compress))
        {
            gzipStream.Write(bytes, 0, bytes.Length);
        }
        
        var compressed = outputStream.ToArray();
        return Convert.ToBase64String(compressed);
    }
}
